sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):
+    def __init__(self, **opts):
+        super().__init__(**opts)
+
+    def send_mail(self, msg):
+        try:
+            return self.client.mail.send.post(request_body=msg.get())
+        except Exception as e:

```

Comment:
That makes sense. BTW, the reference for SendGrid errors [is here](https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/errors.html) and [here](https://github.com/sendgrid/sendgrid-python/tree/0942f9de2d5ba5fedb65a23940ebe1005a21a6c7#error-handling) was how we documented previously.

I think what @w- is trying to say is that this error handling should be done in python-http-client instead, and then here we wrap the errors inside of the custom SendGrid Exception Handlers. Essentially, we would  place [this functionality](https://github.com/sendgrid/sendgrid-python/blob/0942f9de2d5ba5fedb65a23940ebe1005a21a6c7/sendgrid/sendgrid.py#L133), [here](https://github.com/sendgrid/python-http-client/blob/master/python_http_client/client.py#L204). However, the SendGrid specific exception handling would stay in this library.